athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -305,6 +310,46 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
         return new GetSplitsResponse(request.getCatalogName(), splits);
     }
 
+    @Override
+    public GetDataSourceCapabilitiesResponse doGetDataSourceCapabilities(BlockAllocator allocator, GetDataSourceCapabilitiesRequest request)
+    {
+        ImmutableMap.Builder<String, List<OptimizationSubType>> capabilities = ImmutableMap.builder();
+        queryPassthrough.addQueryPassthroughCapabilityIfEnabled(capabilities, configOptions);
+
+        return new GetDataSourceCapabilitiesResponse(request.getCatalogName(), capabilities.build());
+    }
+
+    @Override
+    public GetTableResponse doGetQueryPassthroughSchema(BlockAllocator allocator, GetTableRequest request) throws Exception
+    {
+        logger.debug("doGetQueryPassthroughSchema: enter - " + request);
+        if (!request.isQueryPassthrough()) {
+            throw new IllegalArgumentException("No Query passed through [{}]" + request);
+        }
+
+        String index = request.getQueryPassthroughArguments().get(ElasticsearchQueryPassthrough.INDEX);
+        String endpoint = getDomainEndpoint(request.getQueryPassthroughArguments().get(ElasticsearchQueryPassthrough.SCHEMA));
+        Schema schema = getSchema(index, endpoint);
+
+        return new GetTableResponse(request.getCatalogName(), request.getTableName(),
+                (schema == null) ? SchemaBuilder.newBuilder().build() : schema, Collections.emptySet());
+    }
+
+    private Schema getSchema(String index, String endpoint)
+    {
+        Schema schema;
+        AwsRestHighLevelClient client = clientFactory.getOrCreateClient(endpoint);
+        try {
+            Map<String, Object> mappings = client.getMapping(index);
+            schema = ElasticsearchSchemaUtils.parseMapping(mappings);
+        }
+        catch (IOException error) {
+            throw new RuntimeException("Error retrieving mapping information for index (" +
```

Comment:
can you please not pass the error message from the io Exception, passing it as a cause should be enough.

Reply:
removed.

Reply:
sorry I think you miss understood me; continue to pass the exception as the cause; you just don't need to log the original exception message also; by called e.getmessage();